### ISSUE TITLE
Integrate App Rating Alert

### ISSUE THREAD

**Issue by pacu** [zcash/secant-ios-wallet#265]
Integrate App Store rating component but leave it off with config flag for now.

When to show it:
A. after first sync
B. after first receive of funds
C. A or B when version changes.
D. on request inside debug menu for testing purposes.
TBD: 
@geffenz define the conditions to show the rating view to the user.

**Comment by geffenz** [comment]
The user should have:

- backed up their wallet and passed the test
- sent and received (any amount)

**Comment by LukasKorba** [comment]
I spent some time with the rating dialog and I haven’t been able to figure out any tangible solution for the conditions, reasons?
- the app rating dialog is now controlled by the system (any custom solution is against the guidelines)
- the system decides if it’s presented to the user or not + ensures it won’t happen more than 3 times a year
- there’s no API for us to know if the dialog was presented

that’s for the dialog itself, but even on our side:
- wait for sync + receive of funds condition is also a bit tricky… what if user synced and already has some funds? should I wait for new receive after I marked some timestamp in the user defaults? how do I know there even was a receive without calling getAllTransactions? Such call seems too expensive to do every start of the app..
- let’s say the very first attempt to show the dialog will truly do it.. I mark the timestamp, version and until there’s new version I don’t try to show it again.. but what if there’s new version in a few days and I try to show it not knowing it showed at all? Should I update the version in UD anyway?

With this new policy regarding the dialog, I think we need to clarify:
- what is the secant app update policy going to be? Will we update the app every 2-3 weeks? If yes, we try to show the dialog let’s say once a month…

Being completely blind sucks here, I’m looking for some meaningful but easiest solution…
I think trying to show the dialog 12-times a year seems reasonably easy solution yet still somewhat powerful.. 3 out of those 12 will probably pass and present to the user.. higher frequency looks pointless…

also: 
```
When you call this method while your app is in development mode, a rating and review request view is always displayed so you can test the user interface and experience. However, this method has no effect when you call it in an app that you distribute using TestFlight.
```
that means having `rate the app` in the debug menu probably won't work as expected
according to apple's:
```
Although you normally call this method when it makes sense in the user experience flow of your app, App Store policy governs the actual display of a rating and review request view. Because this method may not present an alert, it isn't appropriate to call requestReview() or requestReview(in:) in response to a button tap or other user action.
```

and from the doc:
```
The SKStoreReviewController API lets you give users an easy way to provide feedback about your app. You can prompt for ratings up to three times in a 365-day period. Users will submit a rating through the standardized prompt, and can write and submit a review without leaving the app.
```
